Fixes #398:-  Prototype for Media wall

Achint08 opened the pull request:
**Changes proposed in this pull request**
- Added necessary reducers and actions for loklak media walls.
  1.  To change Accent Color for media walls
  2.  To change Background Color for media walls
  3. To change Text Color for media walls
  4. To collect all media elements from Api Response to display on grid.
- Added Color customization 
- Added responsive images to loklak media wall
To avoid big PR, adding components in chunks.

Please review @hemantjadon .
**How to test**
Test link :- https://achint08.github.io/loklak_search/
You might find an option to disable "Unsafe scripts" by clicking on the shield/info shown on search bar. Disable it as api is on http. Search for something. You will find an option for media walls  
Click on "Display media wall" to show media wall This is currently a basic prototype for loklak media wall.

**Screenshot**

![screen shot 2017-06-21 at 2 14 07 am](https://user-images.githubusercontent.com/15813932/27355092-5c3de466-5627-11e7-810c-559a46467f29.png)

Fixes #398

Achint08 commented:
@hemantjadon 
 A) Yes. Look at this https://github.com/fossasia/labs.fossasia.org/issues/196 . Customization just for colors. That's it.
 B) For a single on going session, we need a state so that information can be kept.
 C) Yes. It will  be just an additional view.

Achint08 commented:
@hemantjadon Checkout now. A basic prototype of media walls. Any suggestion?

hemantjadon commented on src/app/feed/feed-header/feed-header.component.html:
```diff
@@ -35,4 +35,7 @@
 				</ul>
 			<hr [style.left]="getCoordinates()">
 		</div>
+		<div class="mediaDisplay" *ngIf="(areResultsAvailable.length > 0) && (!resultsLoading)">
```
Make the `areResultsAvailable` as a boolean itself, why its length is checked?

hemantjadon commented on src/app/feed/feed-header/feed-header.component.scss:
```diff
@@ -137,6 +137,12 @@
 			}
 	}
+
+	div.mediaDisplay {
```
Use `kebab-case` for css classes and properties

hemantjadon commented on src/app/reducers/media-wall.ts:
```diff
@@ -0,0 +1,119 @@
+
+
+export interface State {
+	MediaWallsColor;
```
Use `MediaWallColor`, not plural

hemantjadon commented on src/app/media-wall/media-wall.component.scss:
```diff
@@ -0,0 +1,18 @@
+		height:4vh;
+		text-align:center;
+		display:block;
+  	font-family: 'Montserrat', sans-serif !important;
```
Indentation error I guess

hemantjadon commented on src/app/media-wall/media-wall.component.ts:
```diff
@@ -0,0 +1,66 @@
+import { Component,
+					OnInit, AfterViewInit, OnDestroy,
```
Indentation Error Maybe

Achint08 commented:
@djmgit @singhpratyush @kavithaenair @vibhcool Please review :)

mariobehling commented:
Which issue is this related to?

Achint08 commented:
@mariobehling This is a basic prototype of loklak media wall. This is a base for loklak media wall.
All the actions, services are added for loklak media wall to run. It is related to #398 .

hemantjadon commented:
@Achint08 Please fix the merge conflicts then this PR is good to go 👍

mariobehling commented:
@Achint08 Screenshots, Link to issue and link to a test site are not provided. Please follow best practices!

mariobehling commented:
Travis is failing in this PR. Please fix it.

Achint08 commented:
@mariobehling I deleted the test link by mistake. Updated it. Do have a look once.
Check on bigger screens/smaller screens if image adjusts itself to every type of screen.
The travis is failing due to previous PR merged. Looking into that.

mariobehling commented:
We had discussed this feature a number of times and it was mentioned that it should be integrated into the existing application -> as a button "display this search result as a media wall". This is a very simple requirement and we don't need a black page. A simple example is already on the apps page: http://apps.loklak.org/details.html?q=social-wall 

I am unclear why you are not following this.

First step: After the user makes a search, on the search result pags add a text button "Views". When the user clicks on it an option appears "Media Wall". When the user clicks this button another tab opens showing the search results as a media wall in a simple style as http://apps.loklak.org/details.html?q=social-wall
![screenshot from 2017-06-20 23-10-49](https://user-images.githubusercontent.com/1583873/27356344-671aaf68-560e-11e7-87c7-6f98ee15581d.png)

mariobehling commented:
Follow best practices!

> All the actions, services are added for loklak media wall to run. It is related to #398 .

Add this issue to the description!

Achint08 commented:
@mariobehling I have done like that only. Checkout here.
https://achint08.github.io/loklak_search/ .
 

![screen shot 2017-06-21 at 1 27 12 pm](https://user-images.githubusercontent.com/15813932/27373328-82ab33ac-5685-11e7-8868-d31015c06f67.png)

However, my idea about media wall is to give more emphasis on media so it will have a UI like this.

![screen shot 2017-06-21 at 1 30 07 pm](https://user-images.githubusercontent.com/15813932/27373414-d12cbb90-5685-11e7-9dc9-0738fcf40f73.png)

vibhcool commented on src/app/feed/feed-header/feed-header.component.scss:
```diff
@@ -174,4 +180,8 @@
 		left: 20px !important;
```
Could you remove `!important;`, if it is not needed. It seems to be bad practice. And the first answer in this link is too comical 😂  
https://stackoverflow.com/questions/9245353/what-does-important-in-css-mean

mariobehling commented:
@Achint08 It does not make sense if each single feature of the app has a different design. Please follow the design of the app. Implement a simple first version based on the existing work in the FOSSASIA community.
